### PR TITLE
Fix for undefined bundle name with multiple entry points

### PR DESCRIPTION
**Pull request by dbilgili**
Related issue: https://github.com/albinotonnina/parcel-plugin-wrapper/issues/6

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -23,13 +23,16 @@ ${footer}`
 
   const processAsset = async (bundle, processFn) => {
     const {name} = bundle
-    const wrappingCode = await processFn({name, bundler})
 
-    if (wrappingCode) {
-      writeAsset(name, wrappingCode)
+    if (name !== undefined) {
```
Comment (asos-albinotonnina) [inline index.js:27]: Thank you very much, super fast :)
Does this mean that if `bundle.name` is not set the plugin won't do anything?
Is `bundle.name` always undefined when using multiple entry points?
Not following much the development of Parcel right now.
I'm happy to merge this one but could you clarify the above points?
Thank you again

**Review by asos-albinotonnina — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (dbilgili) [reply index.js:27]: Hey Albino,
Sure thing. Having an `undefined` bundle name leads to a catch there.
https://github.com/albinotonnina/parcel-plugin-wrapper/blob/95d7a81bed0b2f36b14b72cfea9baf95e29e6ea8/index.js#L44-L48

As far as I have tested `bundle.name` is always `undefined` when there are multiple entry points. Whereas, `bundle.childBundles` contains these multiple entry points with correct `name` values.

Comment (asos-albinotonnina) [reply index.js:27]: I see, thank you!
I guess traversing the childBundles and applying the plugin to each of the output files could be something to explore?
Feels like a different PR tho. Is it something worth considering anyway?

File: index.js [hunk 2]
```diff
@@ -23,13 +23,16 @@ ${footer}`
...
+      const wrappingCode = await processFn({name, bundler})
+
+      if (wrappingCode) {
+        writeAsset(name, wrappingCode)
+      }
     }
 
-    bundle.childBundles.forEach(function(bundle) {
+    bundle.childBundles.forEach((bundle) => {
       processAsset(bundle, processFn)
```
Comment (jacksteamdev) [inline index.js:36]: @asos-albinotonnina I think @dbilgili is saying that when `processAsset` is called recursively here, `bundle.name` is defined. So the child bundles are traversed in this `forEach` call, right?

File: index.js (hunk 1 above)
Comment (jacksteamdev) [reply index.js:27]: > Is `bundle.name` always undefined when using multiple entry points?

Parcel creates an unnamed container bundle that contains the multiple entry points. It looks like the `childBundles` are the actual entry points.

**Review by jacksteamdev — APPROVED** [review]
I pulled this and can confirm that it's working!

File: index.js (hunk 1 above)
Comment (dbilgili) [reply index.js:27]: > I see, thank you!
> I guess traversing the childBundles and applying the plugin to each of the output files could be something to explore?
> Feels like a different PR tho. Is it something worth considering anyway?

This recursion is already doing it when there are multiple entry points
https://github.com/albinotonnina/parcel-plugin-wrapper/blob/95d7a81bed0b2f36b14b72cfea9baf95e29e6ea8/index.js#L32-L34

And note that calling this line regardless is crucial. For example entry point might be an `index.html` with the following content:

```
<div id="root">
</div>
<script type="text/javascript" src="./index.js"></script>
```

In this case parcel creates an output similar to this:

```
<div id="root"> 
</div><script type="text/javascript" src="/ui.cdfc7045.js"></script>
```

Not traversing `bundle.childBundles` would result in `ui.cdfc7045.js` being omitted.
This is why `forEach` function called on `bundle.childBundles` is not in `else` of the `if` statement, but outside.

File: index.js (hunk 2 above)
Comment (dbilgili) [reply index.js:36]: Exactly.

Comment (asos-albinotonnina) [reply index.js:36]: fine for me then

**Review by asos-albinotonnina — APPROVED** [review]

**Comment by asos-albinotonnina** [comment]
Thanks @dbilgili for the PR and thanks @jacksteamdev for the review :)
I published the new version on npm: https://www.npmjs.com/package/parcel-plugin-wrapper

**Pull request merged**